src/Stratis.Bitcoin.Features.PoA/PoAMiner.cs

```diff
@@ -414,22 +412,20 @@ private async Task<uint> WaitUntilMiningSlotAsync()
                     continue;
                 }
 
-                if (myTimestamp == null)
+                try
                 {
-                    try
-                    {
-                        myTimestamp = this.slotsManager.GetMiningTimestamp(timeNow);
-                    }
-                    catch (NotAFederationMemberException)
-                    {
-                        this.logger.LogWarning("This node is no longer a federation member!");
+                    uint myTimestamp = this.slotsManager.GetMiningTimestamp(timeNow);
+
+                    if (myTimestamp <= timeNow)
+                        return myTimestamp;
 
-                        throw new OperationCanceledException();
-                    }
                 }
+                catch (NotAFederationMemberException)
+                {
+                    this.logger.LogWarning("This node is no longer a federation member!");
 
-                if (myTimestamp <= (timeNow + 1))
```

Comment:
Ohh I see that the +1 was removed here? Was that not needed?

Reply:
I don't think its necessary. I imagine the reasoning was to start the mining of the block 1 second before the timestamp. However the 1 second value seems entirely arbitrary and just introduces inaccuracies when we already allow blocks to be a second or more late. None of this affects the actual timestamp value placed on the block.